projects/buttons_and_adc/src/main.c

```diff
@@ -0,0 +1,61 @@
+// Created by Sudhish M on 2020-10-04.
+// buttons_and_adc
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress adc_pin_address{ .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_address{ .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings button_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = PIO_RES_PULLDOWN,
+  // Reason for using pulldown: when no signal is received
+  // from the button (unpressed), the value sent to the pin is 0
+  .alt_function = GPIO_ALTFN_NONE
+};
+
+static GpioSettings adc_pin_settings = {
+  .direction GPIO_DIR_IN, .resistor = GPIO_RES_NONE, .alt_function = GPIO_ALTFN_ANALOG
+};
+
+static InterruptSettings interrupt_settings =
+    {
+      .type = INTERRUPT_TYPE_INTERRUPT,
+      .priority = INTERRUPT_PRIORITY_NORMAL,
+    }
+
+static void
+prv_callback(const GpioAddress *address, void *context) {
```

Comment:
`make format` should take care of this, but we write the type and qualifiers on the same line as the function declaration, like `static void prv_callback(...) {`.